/ros2_laser_merger:
  ros__parameters:
    # 병합 후 PointCloud 퍼블리시 토픽 및 기준 프레임 설정
    pointCloudTopic: cloud_in
    pointCloutFrameId: base_scan
    qos_overrides./parameter_events.publisher.depth: 1000 #1000
    qos_overrides./parameter_events.publisher.reliability: best_effort #best_effort
    use_sim_time: false               #false

    # scan0 (왼쪽 라이다)
    scanTopic1: /scan0         # 앞쪽 라이다의 LaserScan 토픽 이름
    laser1XOff: 0.0                 # base_scan 기준 x 오프셋
    laser1YOff: 0.282                # base_scan 기준 y 오프셋
    laser1ZOff: 0.0                   # base_scan 기준 z 오프셋 (이미 base_scan 프레임이 기준임)
    laser1Alpha: 0.0                # 회전 각: -45°
    laser1AngleMin: 135.0             # -138 / 스캔 각 최소값 (필요 시 조정)
    laser1AngleMax: 225.0             # +138 / 스캔 각 최대값 (필요 시 조정)
    laser1R: 255                      # 색상 (빨간색)
    laser1G: 0
    laser1B: 0
    show1: true
    flip1: false                      # 라이다가 정방향으로 달려져 있으므로
    inverse1: true

    # scan1 (오른쪽 라이다)
    scanTopic2: /scan1          # 뒷쪽 라이다의 LaserScan 토픽 이름
    laser2XOff: 0.0               # base_scan 기준 x 오프셋
    laser2YOff: -0.282                  # base_scan 기준 y 오프셋
    laser2ZOff: 0.0                   # base_scan 기준 z 오프셋
    laser2Alpha: 0.0                # 회전 각: 135°
    laser2AngleMin: 135.0             # -138 / 스캔 각 최소값
    laser2AngleMax: 225.0             # +138 / 스캔 각 최대값
    laser2R: 0                     	  # 색상 (파란색)
    laser2G: 0
    laser2B: 255
    show2: true
    flip2: false                       # 라이다가 정방향으로 달려져 있으므로
    inverse2: true

/pointcloud_to_laserscan:
  ros__parameters:
    angle_increment: 0.0058 #0.0058 ->0.33도
    angle_max: 2.36
    angle_min: -2.36
    inf_epsilon: 1.0
    max_height: 1.0
    min_height: 0.0
    qos_overrides:
      /parameter_events:
        publisher:
          depth: 1000 #1000
          reliability: best_effort #best_effort
    queue_size: 16
    range_max: 25.0
    range_min: 0.45
    scan_time: 0.03333333333333333
    target_frame: base_scan
    transform_tolerance: 0.01 #0.01
    use_inf: true
    use_sim_time: false #false
